detect_meteors.py

```diff
@@ -176,11 +182,52 @@ def detect_meteors(rf_dir, id_dir, noise_dir, output_dir,
         tmm.loadFromHdf5('/tmp/tmm.hdf5', skip_lowlevel=True)
     else:
         tmm.loadFromHdf5(skip_lowlevel=True)
-
-    for k, (tx, rx) in enumerate(data_generator(rfo, ido, no, tmm, s0, s1, rxch, txch)):
-        #FIXME call processing functions here
-        pass
-
+    
+    data_list = []
+    saved_data = np.zeros((59858, 1120))
+    pulse_data = data_generator(rfo, ido, no, tmm, s0, s1, rxch, txch)
+    times = []
+    for k, (tx, rx) in enumerate(pulse_data):
+
+        num_of_pulses = np.int((s1 - s0)/(rx.sample_rate*0.0267300605774))
+        # function that calculates snr values
+        def data_cal(array):
+            snr_vals = (np.abs(array.values)**2)/rx.noise_power
+            snr_point = np.unravel_index(np.argmax(snr_vals), (6879, 480))
+            max_snr = snr_vals[snr_point[0], snr_point[1]]
+            return max_snr, snr_point, snr_vals
```

Comment:
This looks a lot like the valargmax function I made at one time, so you can look to that for reference. The SNR calculation should be taken outside of this function and put on its own line. Then the function stands alone, and it can go in the module with the other meteor processing functions.

The way it is currently defined, the function is created every loop iteration. This can be expensive. It also violates the convention that everything the function needs to operate appears in its argument list, which can be confusing and therefore a place where bugs will happen.